content/scalability/docs/workloads.md

```diff
@@ -0,0 +1,87 @@
+# Workloads
```

Comment:
Could you look into adding callouts object size and object count, and how its impact etcd storage consumed

a. object count -  As total object count  increase the etcd db size also increases. Kubernetes publishes a metric to show total count of objects in etcd per object kind. 
Prior to kubernetes version 1.22, use etcd_object_count metric. 
Above 1.22, metric name is apiserver_storage_object_counts.

Query for 1.22 and above

kubectl get --raw=/metrics | grep apiserver_storage_object_counts |awk '$2>100' |sort -n -k 2


Query for clusters below 1.22

kubectl get --raw=/metrics | grep etcd_object_counts |awk '$2>100' |sort -n -k 2

2. Object size - Sometimes, the size of the object such as a pod spec can be optimized to avoid hitting the db quota. For example instead of embedding any big blob inside each pod spec,  it can be referenced. This can be done by mounting the blob using configmap or secret and referencing that from each pod. This brings down the pod spec size significantly. 
[Upstream example](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/) (https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/) shows how to reference a secret from a pod. 

To look at the object size, run kubectl get to dump out the object in a yaml and notice the size of that yaml.
Example:

kubectl get pod test -o yaml > /tmp/l

and then do 

ls -l /tmp/l (to see the size)

Reply:
We left out a lot of etcd and monitoring advice because that is our responsibility so long as customers are not abusing the functionality. object size was the main reason to add the webhooks section under workloads. Do you think there's a behavior that causes large size or too many objects that is actionable for customers besides monitoring it?

Reply:
I am working with the etcd team on a blog on what customer can do when they overrun etcd db size. Today, it is partly actionable. let me explain - if customer has already overrun the 8GB size, then the etcd becomes read only, and the customer would need to engage the EKS etcd team to take some action , before it become read/write and customer can delete the objects. On the other hand, if the etcd is approaching the DB size (but not fully consumed), then customer's can delete the object proactively. So, there is value in monitoring those.  On the roadmap, we have an item to let customers  take an action even when the DB is fully consumed (without service team involvement but it is not there today).

Reply:
I agree and while writing this doc I realized we need another doc for monitoring best practices (including etcd). I don't think it fits here, but would love to start outlining monitoring best practices with you and support. There's a lot of info we can provide and we can work with other teams (Cloudwatch and grafana) to make better default dashboards and alerts for customers.

Reply:
Sounds good. I like the idea of monitoring best practices. Happy to collaborate with you on that after we get this out of the door 👍